solr/core/src/java/org/apache/solr/api/V2HttpCall.java

```diff
@@ -174,7 +174,7 @@ public void call(SolrQueryRequest req, SolrQueryResponse rsp) {
             }
           }
         }
-      } else if ("cores".equals(prefix)) {
+      } else if ("cores".equals(prefix) && pathSegments.size() > 1) {
         origCorename = pathSegments.get(1);
```

Comment:
This is getting an index OOB if you request `/cores`, could combine the size check with the existing size check on the earlier if statement...